src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs

```diff
@@ -20,10 +20,12 @@ public abstract partial class JSRuntime : IJSRuntime
     {
         private long _nextObjectReferenceId; // Initial value of 0 signals no object, but we increment prior to assignment. The first tracked object should have id 1
         private long _nextPendingTaskId = 1; // Start at 1 because zero signals "no response needed"
-        private readonly ConcurrentDictionary<long, object> _pendingTasks = new ConcurrentDictionary<long, object>();
-        private readonly ConcurrentDictionary<long, IDotNetObjectReference> _trackedRefsById = new ConcurrentDictionary<long, IDotNetObjectReference>();
-        private readonly ConcurrentDictionary<long, CancellationTokenRegistration> _cancellationRegistrations =
-            new ConcurrentDictionary<long, CancellationTokenRegistration>();
+        private readonly ConcurrentDictionary<long, object> _pendingTasks = new();
+        private readonly ConcurrentDictionary<long, IDotNetObjectReference> _trackedRefsById = new();
+        private readonly ConcurrentDictionary<long, CancellationTokenRegistration> _cancellationRegistrations = new();
+
+        internal readonly ArrayBuilder<byte[]> ByteArraysToBeRevived = new();
```

Comment:
`ArrayBuilder<T>` needs to be disposed otherwise it's nonempty, it won't return its final array to the underlying pool.

Perhaps you'll need to make `JSRuntime` itself also disposable to handle this. I can't immediately anticipate any further complexity that would cause but if it did end up being problematic we could think about ways to not use a pooled store for `ByteArraysToBeRevived`.